clis/teliod/src/qnap.rs

```diff
@@ -0,0 +1,398 @@
+use std::{
+    fs, io,
+    path::PathBuf,
+    process::{Command, Stdio},
+    time::Duration,
+};
+
+use const_format::concatcp;
+use rust_cgi::{http::Method, text_response, Request, Response};
+use tracing::level_filters::LevelFilter;
+use uuid::Uuid;
+
+use crate::{
+    command_listener::CommandResponse,
+    config::{InterfaceConfig, MqttConfig, TeliodDaemonConfig},
+    ClientCmd, DaemonSocket, TIMEOUT_SEC,
+};
+
+const TELIOD_TMP_DIR: &str = "/tmp/nordsecuritymeshnet";
+const PID_FILE: &str = concatcp!(TELIOD_TMP_DIR, "/teliod.pid");
+const QPKG_DIR: &str = "/share/CACHEDEV1_DATA/.qpkg/NordSecurityMeshnet";
+const TELIOD_BIN: &str = concatcp!(QPKG_DIR, "/teliod");
+const MESHNET_LOG: &str = concatcp!(QPKG_DIR, "/meshnet.log");
+const TELIOD_LOG: &str = "/var/log/teliod.log";
+
+#[cfg(not(test))]
+const TELIOD_CFG: &str = concatcp!(QPKG_DIR, "/teliod.cfg");
+#[cfg(test)]
+use tests::TELIOD_CFG;
+
+pub(crate) fn handle_request(request: Request) -> Response {
+    match (request.method(), request.uri().query()) {
+        (&Method::POST, _) => start_daemon(),
+        (&Method::DELETE, _) => stop_daemon(),
+        (&Method::PATCH, _) => {
+            let body = match String::from_utf8(request.into_body()) {
+                Ok(body) => body,
+                Err(_) => return text_response(400, "Invalid UTF-8 in request body."),
+            };
+            update_config(&body)
+        }
+        (&Method::GET, Some("info=get-status")) => get_status(),
```

Comment:
Will you add proper query parsing on next PR ?

Reply:
yes, I'll leave it for the next PR